diagnostic_common_diagnostics/src/diagnostic_common_diagnostics/sensors_monitor.py

```diff
@@ -156,13 +159,16 @@ def _rpm_to_rads(rpm):
 
 
 def parse_sensors_output(output):
-    out = StringIO(output.decode('utf-8'))
+    out = StringIO(output if isinstance(output, str) else output.decode('utf-8'))
 
     sensorList = []
     for line in out.readlines():
         # Check for a colon
         if ":" in line and "Adapter" not in line:
-            s = parse_sensor_line(line)
+            try:
+                s = parse_sensor_line(line)
+            except ValueError:
```

Comment:
I would change this line to:
`
try:
    s = parse_sensor_line(line)
except Exception as exc:
   rospy.logwarn('Unable to parse line "%s", due to %s', line, exc)
`
(please use this type of formatting with ROS to avoid formatting the string if the chosen loglevel is disabled=